In [1]:
%load_ext sql

In [2]:
%sql postgresql://postgres:0000@localhost/postgres

In [9]:
%%sql
create table Titanic (PassengerId int primary key, Survived int, Pclass int, Name varchar(100),Sex varchar(10), Age float ,SibSp int, Parch int, Ticket varchar(100), Fare float, Cabin varchar(100), Embarked varchar(100));
copy titanic from '/home/natfiik/Documents/SAE_BDD/titanic_train.csv' delimiter ',' CSV header;

 * postgresql://postgres:***@localhost/postgres
Done.
891 rows affected.


[]

In [10]:
%%sql

Create table Survivant(PassengerId int, Survived int, pclass int, name varchar(100) primary key, sex varchar(10), age float, foreign key (passengerid) references Titanic(passengerId));
Create table Ticket (PassengerId int primary key, ticket varchar(100),fare float, cabin varchar(30), embarked varchar(100), foreign key (Passengerid) references titanic(Passengerid));
Create table Classe (PassengerId int primary key, survived int, Pclass int, foreign key(PassengerId) references titanic(PassengerId));
Create table Enregistrement (PassengerId int, name varchar(100), embarked varchar(10), Pclass int,primary key(PassengerId, Name), foreign key (passengerid) references titanic(passengerid));
Create table Famille (PassengerId int, name varchar(100) primary key, SibSp int, Parch int,foreign key (passengerId) references Titanic(passengerid));

 * postgresql://postgres:***@localhost/postgres
Done.
Done.
Done.
Done.
Done.


[]

In [11]:
%%sql

insert into Survivant(PassengerId, Survived,pclass, name, sex, age)
select PassengerId, Survived, Pclass, Name, Sex, Age from Titanic;

insert into Ticket(PassengerId, ticket, fare,cabin,embarked)
select PassengerId, ticket, fare, cabin, embarked from Titanic;

insert into Classe (PassengerId, survived,Pclass)
select PassengerId, survived, Pclass from Titanic;

insert into Enregistrement (passengerid, name, embarked, Pclass)
select PassengerId, Name, Embarked, PClass from Titanic;

insert into Famille (PassengerId, Name, SibSp, Parch)
select PassengerId, Name, SibSp, Parch from Titanic;

 * postgresql://postgres:***@localhost/postgres
891 rows affected.
891 rows affected.
891 rows affected.
891 rows affected.
891 rows affected.


[]

In [8]:
%%sql
drop table Enregistrement;
drop table classe;
drop table ticket;
drop table survivant;
drop table Famille;
drop table Titanic;

 * postgresql://postgres:***@localhost/postgres
Done.
Done.
Done.
Done.
Done.
Done.


[]

# Rapport :

Nous avons créer une première table "Titanic" qui contient toutes les données du fichier csv pour pouvoir ensuite d'alimenter les autres tables à venir.

La première table que nous avons créé est la table "Survivant". Cette table permet de voir qui est le passager, ses informations personnelles et de savoir si il a survécu.

Pour ce faire, nous avons utilisés :
    
    -PassengerId,Survived et Pclass qui sont des integers
    -Name et sex qui sont des varchar
    -Age qui est un float

La seconde table, Ticket, nous permet de récolter les informations du ticket d'un passager.
Pour ce faire, nous avons utilisé les attributs suivant :
    
    -PassengerId, un integer
    - ticket, embarked et cabin qui sont des varchar
    - fare qui est un float
    
La troisième table, Classe, permet de lié les passagers à leurs classes, et si ils ont survécu.
Pour ce faire, nous avons utilisé les attributs suivant :
    
    -PassengerId, Survived et Pclass qui sont des integer
    
La quatrième table, Enregistrement, permet de recruillir les informations qui ont pu être donnée lors de l'achat d'un billiet.
Pour ce faire, nous avons utilisé les atributs suivant :
    
    -PassengerId, Pclass qui sont des integer.
    -Name, embarked qui sont des varchar
    
Finalement, la table Famille nous permet de savoir pour chaque passager si il a de la famille, et si il en a, le nombre de frère,soeur et epoux, de parent et d'enfant.
Pour ce faire, nous avons utilisé les attributs suivant :
    
    -PassengerId, SisSp et Parch sont des integer
    -Name est une varchar de 100 charactères

#    Statistiques :

Il y a 3 classes différentes : 

    En première classe, il y a 216 personnes composé de 122 hommes et 94 femmes.
    En deuxième classe, il y a 184 personnes composé de 108 hommes et 76 femmes.
    En troisième classe, il y a 491 personnes composé de 347 hommes et 144 femmes.


Parmi les 891 passagers du Titanic :

    Seulement 342 personnes ont survécu, et 549 sont décédées. 
    Il y a donc seulement environ 38% de survivants pour 62% de morts

    Sur les 314 femmes présentes, 233 ont survécues et 81 sont mortes (25,7% sont mortes)
    
    Sur les 577 hommes présents, 109 ont survécus et 468 sont morts (81% sont morts)
    
    
    
    
    Pour la première classe, nous avons :
        - 3 femmes mortes pour 91 femmes vivantes (3% sont mortes)
        - 77 hommes morts pour 45 femmes vivants (63% sont morts)
    
    Pour la deuxième classe, nous avons :
        - 6 femmes mortes pour 70 femmes vivantes (7.8% sont mortes)
        - 91 hommes morts pour 17 hommes vivants (84% sont morts)
        
    Pour la troisième classe, nous avons :
        - 72 femmes mortes pour 72 femmes vivantes (50% sont mortes)
        - 300 hommes morts pour 47 hommes vivants (86% sont morts)
    
# Voici les différentes requêtes effectuées :

### Nombre de classes de passager :

In [12]:
%%sql
select count(distinct pclass) from Classe;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(3,)]

### Nombre de passagers par classe :

In [13]:
%%sql
select count(PassengerId) from Enregistrement where Pclass= 1;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(216,)]

In [14]:
%%sql
select count(PassengerId) from Enregistrement where Pclass= 2;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(184,)]

In [15]:
%%sql
select count(PassengerId) from Enregistrement where Pclass= 3;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(491,)]

In [73]:
engine = "postgresql://postgres:0000@localhost/postgres"
df1 = pandas.read_sql("select pclass,count(pclass) from Classe group by pclass",engine)
fig = px.bar(df1, x="pclass", y="count",title="Nombre de passagers par classe")
fig.show()

### Nombre d'hommes et de femmes par classe :

#### Pour la classe 1 :

In [16]:
%%sql
select count(passengerId) from Survivant where sex='male' and pclass = 1

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(122,)]

In [17]:
%%sql
select count(passengerId) from Survivant where sex='female' and pclass = 1

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(94,)]

#### Pour la classe 2 :


In [18]:
%%sql
select count(passengerId) from Survivant where sex='male' and pclass = 2

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(108,)]

In [19]:
%%sql
select count(passengerId) from Survivant where sex='female' and pclass = 2

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(76,)]

#### Pour la classe 3 :

In [20]:
%%sql
select count(passengerId) from Survivant where sex='male' and pclass = 3

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(347,)]

In [21]:
%%sql
select count(passengerId) from Survivant where sex='female' and pclass = 3

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(144,)]

In [70]:
engine = "postgresql://postgres:0000@localhost/postgres"
df1 = pandas.read_sql("select pclass,sex,count(sex) from Survivant group by pclass,sex",engine)
fig = px.bar(df1, x="pclass", y="count",color = "sex",title="Nombre d'hommes et femmes par classe")
fig.show()

### Nombre de survivants et de morts :

In [22]:
%%sql
select count(PassengerId) from survivant where survived = 1

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(342,)]

In [23]:
%%sql
select count(PassengerId) from survivant where survived = 0

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(549,)]

In [66]:
engine = "postgresql://postgres:0000@localhost/postgres"
df1 = pandas.read_sql("select survived, count(survived) from Survivant group by survived",engine)
fig = px.bar(df1, x="survived", y="count",color = "survived",title="Survie des passagers")
fig.show()

### Nombre de morts et survivants par classe :

#### Pour la classe 1 :

In [24]:
%%sql
select count(passengerId) from Classe where pclass = 1 and survived = 1

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(136,)]

In [25]:
%%sql
select count(passengerId) from Classe where pclass = 1 and survived = 0

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(80,)]

#### Pour la classe 2 :


In [26]:
%%sql
select count(passengerId) from Classe where pclass = 2 and survived = 1

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(87,)]

In [27]:
%%sql
select count(passengerId) from Classe where pclass = 2 and survived = 0

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(97,)]

   #### Pour la classe 3 :

In [28]:
%%sql
select count(passengerId) from Classe where pclass = 3 and survived = 1

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(119,)]

In [29]:
%%sql
select count(passengerId) from Classe where pclass = 3 and survived = 0

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(372,)]

In [63]:
engine = "postgresql://postgres:0000@localhost/postgres"
df1 = pandas.read_sql("select pclass,survived,count(PassengerId) from titanic group by pclass,survived order by survived desc",engine)
fig = px.bar(df1, x="pclass", y="count",color = "survived",title="Survie des passagers par classe")
fig.show()

### Nombre de morts et survivants par sexe :

#### Pour les femmes :

In [30]:
%%sql
select count(passengerId) from survivant where sex ='female' and survived = 1;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(233,)]

In [31]:
%%sql
select count(passengerId) from survivant where sex ='female' and survived = 0

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(81,)]

#### Pour les hommes :

In [32]:
%%sql
select count(passengerId) from survivant where sex ='male' and survived = 1;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(109,)]

In [33]:
%%sql
select count(passengerId) from survivant where sex ='male' and survived = 0;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(468,)]

In [64]:
engine = "postgresql://postgres:0000@localhost/postgres"
df1 = pandas.read_sql("select sex,survived,count(PassengerId) from titanic group by sex,survived order by survived desc",engine)
fig = px.bar(df1, x="sex", y="count",color = "survived",title="Survie par sexe")
fig.show()

###  Nombre de morts et survivants par sexe et classe :

   #### Pour la classe 1 :
    Femmes :

In [34]:
%%sql
select count(passengerId) from survivant where sex ='female' and pclass = 1 and survived = 1;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(91,)]

In [35]:
%%sql
select count(passengerId) from survivant where sex ='female' and pclass = 1 and survived = 0;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(3,)]

    Hommes :

In [36]:
%%sql
select count(passengerId) from survivant where sex ='male' and pclass = 1 and survived = 1;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(45,)]

In [37]:
%%sql
select count(passengerId) from survivant where sex ='male' and pclass = 1 and survived = 0;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(77,)]

#### Pour la classe 2 :
    Femmes :

In [38]:
%%sql
select count(passengerId) from survivant where sex ='female' and pclass = 2 and survived = 1;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(70,)]

In [39]:
%%sql
select count(passengerId) from survivant where sex ='female' and pclass = 2 and survived = 0;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(6,)]

    Hommes :

In [40]:
%%sql
select count(passengerId) from survivant where sex ='male' and pclass = 2 and survived = 1;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(17,)]

In [41]:
%%sql
select count(passengerId) from survivant where sex ='male' and pclass = 2 and survived = 0;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(91,)]

#### Pour la classe 3 :
    Femmes :

In [42]:
%%sql
select count(passengerId) from survivant where sex ='female' and pclass = 3 and survived = 1;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(72,)]

In [43]:
%%sql
select count(passengerId) from survivant where sex ='female' and pclass = 3 and survived = 0;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(72,)]

    Hommes :

In [44]:
%%sql
select count(passengerId) from survivant where sex ='male' and pclass = 3 and survived = 1;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(47,)]

In [45]:
%%sql
select count(passengerId) from survivant where sex ='male' and pclass = 3 and survived = 0;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(300,)]

In [61]:
engine = "postgresql://postgres:0000@localhost/postgres"
df1 = pandas.read_sql("select sex,survived,pclass,count(PassengerId) from titanic where sex = 'female' group by sex,survived,pclass order by survived",engine)
fig = px.bar(df1, x="pclass", y="count",color = "survived",title="Survie des femmes par classe")
fig.show()

In [62]:
engine = "postgresql://postgres:0000@localhost/postgres"
df1 = pandas.read_sql("select sex,survived,pclass,count(PassengerId) from titanic where sex = 'male' group by sex,survived,pclass order by survived",engine)
fig = px.bar(df1, x="pclass", y="count",color = "survived",title="Survie des hommes par classe")
fig.show()